core/src/main/java/com/linecorp/armeria/client/Endpoint.java

```diff
@@ -149,13 +149,17 @@ private static Endpoint create(String host, int port, boolean validateHost) {
             } else {
                 ipV6Addr = host;
             }
-            return new Endpoint(ipV6Addr, ipV6Addr, port, DEFAULT_WEIGHT, HostType.IPv6_ONLY, null);
+            return new Endpoint(ipV6Addr, ipV6Addr, port, DEFAULT_WEIGHT, HostType.IPv6_ONLY, null, false);
         }
 
+        boolean hasTrailingDot = false;
         if (validateHost) {
+            if (host.endsWith(".")) {
+                hasTrailingDot = true;
+            }
             host = InternetDomainName.from(host).toString();
         }
-        return new Endpoint(host, null, port, DEFAULT_WEIGHT, HostType.HOSTNAME_ONLY, null);
+        return new Endpoint(host, null, port, DEFAULT_WEIGHT, HostType.HOSTNAME_ONLY, null, hasTrailingDot);
```

Comment:
Not strong on this, but `hasTrailingDot` seems to imply that `host` ends with a dot. (it is possible that `host` ends with a dot but `hasTrailingDot == false` if `validateHost == false`

```suggestion
        return new Endpoint(host, null, port, DEFAULT_WEIGHT, HostType.HOSTNAME_ONLY, null, implicitTrailingDot);
```

Reply:
I removed `hasTrailingDot()` getter and the field name has been changed to `wasTrailingDotStripped`. The stripped training dot will be recovered when it is converted to a `SocketAddress` via `.toSocketAddress(int)`.